receiver/datadogreceiver/internal/translator/traces_translator.go

```diff
@@ -195,14 +223,14 @@ func tagsToSpanLinks(tags map[string]string, dest ptrace.SpanLinkSlice) error {
 		link := dest.AppendEmpty()
 
 		// Convert trace id.
-		rawTrace, errTrace := trace.TraceIDFromHex(span.TraceID)
+		rawTrace, errTrace := oteltrace.TraceIDFromHex(span.TraceID)
 		if errTrace != nil {
```

Comment:
Ditto, shall we enrich the error message with the provided `hexTraceID` to troubleshoot the error `hex encoded trace-id must have length equals to 32`?

Reply:
Added a wrapping here and on the call to SpanIDFromHex below as well (not related to this PR). This gets me to wonder whether span links will work with the reconstructed spanIDs, I'll need to test that.